crates/primitives/simulations/src/lib.rs

```diff
@@ -1,16 +1,37 @@
+use blockifier::state::errors::StateError;
+use blockifier::transaction::errors::TransactionExecutionError;
 use blockifier::transaction::objects::TransactionExecutionInfo;
+use starknet_api::core::ContractAddress;
 use starknet_core::types::{SimulationFlag, SimulationFlagForEstimateFee};
 
 // TODO: This is a placeholder
 // https://github.com/starkware-libs/starknet-specs/blob/master/api/starknet_api_openrpc.json#L3919
 // The official rpc expect use to return the trace up to the point of failure.
 // Figuring out how to get that is a problem for later
-#[derive(Debug)]
+#[derive(Debug, Clone)]
 #[cfg_attr(feature = "scale-info", derive(scale_info::TypeInfo))]
 #[cfg_attr(feature = "parity-scale-codec", derive(parity_scale_codec::Encode, parity_scale_codec::Decode))]
-pub struct PlaceHolderErrorTypeForFailedStarknetExecution;
+pub enum Error {
```

Comment:
should we rename this to SimulationError to make it more readable when it's imported to other files?